### ISSUE TITLE
Problem with axios instance and getUri

### ISSUE THREAD

**Issue by Midnighter** [axios/axios#2174]
### Describe the bug

I would like to use the new `getUri` method on an axios instance. However, when I attempt this I get a type error.

```
TypeError: Cannot read property 'indexOf' of undefined
```

```
buildURL
node_modules/axios/lib/helpers/buildURL.js:56
  53 | }
  54 | 
  55 | if (serializedParams) {
> 56 |   url += (url.indexOf('?') === -1 ? '?' : '&') + serializedParams;
     | ^  57 | }
  58 | 
  59 | return url;
```

### To Reproduce

The following works:

```js
import axios from "axios";

export const getFulltextURL = reference => {
  return axios.getUri({
    url: "http://localhost",
    params: {
      "url_ver": "Z39.88-2004",
      "url_ctx_fmt": "info:ofi/fmt:kev:mtx:ctx",
      "ctx_ver": "Z39.88-2004",
      "ctx_enc": "info:ofi/enc:UTF-8",
      "rft_genre": "article",
      "rft_val_fmt": "info:ofi/fmt:kev:mtx:journal",
      "rft.doi": reference.doi[0],
      "rft.issn": reference.issn[0]
    }
  });
};
```

but with an axios instance it does not:

```js
import axios from "axios";

const fulltextResolver = axios.create({
  baseURL: "http://localhost",
  params: {
    "url_ver": "Z39.88-2004",
    "url_ctx_fmt": "info:ofi/fmt:kev:mtx:ctx",
    "ctx_ver": "Z39.88-2004",
    "ctx_enc": "info:ofi/enc:UTF-8",
    "rft_genre": "article",
    "rft_val_fmt": "info:ofi/fmt:kev:mtx:journal",
  }
});

export const getFulltextURL = reference => {
  return fulltextResolver.getUri({
    params: {
      "rft.doi": reference.doi[0],
      "rft.issn": reference.issn[0]
    }
  });
};
```

I have also tried creating the instance with `url` and `baseUrl` instead but then I get a different type error.

### Expected behavior

I expect the second example to return the same URL with the parameters of the instance and the method call being merged.

### Environment

 - Axios Version 0.19.0-beta.1
 - OS: Ubuntu 18.04
 - Browser Chromium, Firefox
 - Browser Version 73.0.3683.86, Firefox 67.0 (on Firefox `buildUrl` is undefined)
- React 16.8.6

**Comment by jasonsaayman** [comment]
@Midnighter this seems to relate to #2190, please let me know if that issue is addressing the same issue, there is a pull request for that issue.

**Comment by Alanscut** [comment]
@Midnighter   `url` is a required field, you can verify if based on #2196  and #2555 can meet your needs?

> ```
> TypeError: Cannot read property 'indexOf' of undefined
> ```

**Comment by Midnighter** [comment]
I will give it a try over the weekend.

**Comment by pvsiddhartha** [comment]
Following.
I encountered the similar issue.


```
buildURL.js:62 Uncaught (in promise) TypeError: Cannot read property 'indexOf' of undefined
    at buildURL (buildURL.js:62)
    at dispatchXhrRequest (xhr.js:30)
    at new Promise (<anonymous>)
    at xhrAdapter (xhr.js:12)
    at dispatchRequest (dispatchRequest.js:52)
```

**Comment by jedmonsanto** [comment]
@pvsiddhartha i encountered the same problem and to found out that my url is invalid.

**Comment by Midnighter** [comment]
@Alanscut I have made [a code sandbox](https://codesandbox.io/s/mutable-cookies-d7qmu?file=/src/index.js) and noticed a few things:

1. `url` is a required key in the config. I can work around that by passing an empty string.
2. As noted in #2196 the parameters do not yet get merged (version 0.19.2).
3. As described in #2555, I do not get the full URI.
4. The originally reported error has disappeared :slightly_smiling_face:
